src/Tasks/Microsoft.Managed.After.targets

```diff
@@ -50,21 +50,26 @@ Copyright (C) Microsoft Corporation. All rights reserved.
     <ProjectReferenceTargetsForRebuildInOuterBuild>$(ProjectReferenceTargetsForCleanInOuterBuild);$(ProjectReferenceTargetsForBuildInOuterBuild);$(ProjectReferenceTargetsForRebuildInOuterBuild)</ProjectReferenceTargetsForRebuildInOuterBuild>
     <ProjectReferenceTargetsForRebuild>$(ProjectReferenceTargetsForClean);$(ProjectReferenceTargetsForBuild);$(ProjectReferenceTargetsForRebuild)</ProjectReferenceTargetsForRebuild>
 
-    <ProjectReferenceTargetsForPublish>GetCopyToPublishDirectoryItems;$(ProjectReferenceTargetsForPublish)</ProjectReferenceTargetsForPublish>
+    <!-- Publish chains Build, so inherit it -->
+    <ProjectReferenceTargetsForPublishInOuterBuild>$(ProjectReferenceTargetsForBuildInOuterBuild);$(ProjectReferenceTargetsForPublishInOuterBuild)</ProjectReferenceTargetsForPublishInOuterBuild>
```

Comment:
Are there optimization opportunities we might be missing if we call build explicitly here, and something else could have determined that we could skip building? Like I think there's some switch you can pass to publish to say that you've already built, and we shouldn't build again.

Reply:
Team triage (because ^ was a bit unclear):
There's normally a `dotnet publish --no-build`. With this, that wouldn't be an option for graph builds. Is that something you intend, or were you just assuming people always build before publishing? If that's intentional, we're ok with it, to be clear.

Reply:
Yea, this was something I was thinking about trying to address, but it's difficult. The `NoBuild` property isn't always set on the referencing project; it's sometimes passed as an additional property to the referenced projects (eg sfproj).

However, I do think we can handle `--no-build`, so at least that scenario will be covered.

Reply:
Ok, tested with both `/t:Publish /bl /graph /isolate` and `/t:Publish /bl /graph /isolate /p:NoBuild=true` and both seem to work now.